workbench-core/accounts/src/services/projectService.ts

```diff
@@ -56,20 +62,143 @@ export default class ProjectService {
     }
   }
 
+  // TODO--delete after dynamic Authz
+  private _mockGetUserGroups(): string[] {
+    return [];
+  }
+
   /**
    * List projects
    *
+   * @param request - the optional fields to list projects
+   * @param user - authenticated user listing the projects
    * @returns Project entries in DDB
    */
-  public async listProjects(): Promise<{ data: Project[] }> {
-    const queryParams = {
+  // TODO--add filter support
+  public async listProjects(
+    user: AuthenticatedUser,
+    request?: ListProjectsRequest
+  ): Promise<{ data: Project[]; paginationToken: string | undefined }> {
+    const pageSize = request ? request.pageSize : undefined;
```

Comment:
nit: `request?.pageSize` . Same below

Reply:
Additionally can be  ?? 0 to avoid null check later

Reply:
Changed the code to require `request` to be defined so I think `pageSize = request.pageSize` will work because it will be `undefined` if the optional attribute of `request` does not exist.